### ISSUE TITLE
Multiple llms_calculate_course_data wp-cron events scheduled for the same course.

### ISSUE THREAD

**Issue by pondermatic** [gocodebox/lifterlms#1600]
### Reproduction Steps

I don't know what the steps are, but it is possible for there to be two `llms_calculate_course_data` cron events scheduled for the same course ID. One has the course ID as an integer and the other as a string.

### Expected Behavior

When `LLMS_Processor_Course_Data->schedule_calculation()` prepares the event's arguments, it should cast the course ID to an integer.

### Actual Behavior

1. If a call to `LLMS_Processor_Course_Data->schedule_calculation()` is given a string course ID, an event is scheduled.
1. If a 2nd call to `LLMS_Processor_Course_Data->schedule_calculation()` is given an integer course ID, a **2nd event** is scheduled.

This wastes system resources doing the same calculations twice.

This issue has been recreated:
+ [ ] Locally
+ [ ] On a staging site
+ [x] On a production website
+ [ ] With only LifterLMS and a default theme

**Comment by thomasplevy** [comment]
@pondermatic thanks, I'll have a closer look at fixing this later today but possibly not until next week.

Before I start working on this class again do you have anything else you're going to be reporting regarding this class?

**Comment by pondermatic** [comment]
I'm looking into why the same task (course and page) shows up in multiple batches. It seems that tasks are added to a batch, which creates a new batch.

**Comment by thomasplevy** [comment]
@pondermatic what do you mean "course & page"? A page post type should not be being passed into this class for any reason...

**Comment by pondermatic** [comment]
In the `wp_options` table, I have 
`llms_background_process_course_data_batch_49929a7ab6937b11bad017` = `a:1:{i:0;a:5:{s:7:"post_id";s:5:"12926";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}}`

`llms_background_process_course_data_batch_2bd6ccdf7c02c959dac0d0` = `a:2:{i:0;a:5:{s:7:"post_id";s:5:"12926";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}i:1;a:5:{s:7:"post_id";s:5:"13076";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}}`

`llms_background_process_course_data_batch_ec0b0ed1824c59b8a2e09d` = `a:3:{i:0;a:5:{s:7:"post_id";s:5:"12926";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}i:1;a:5:{s:7:"post_id";s:5:"13076";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}i:2;a:5:{s:7:"post_id";s:5:"13083";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}}`

etc.

Each row contains a batch. Each batch contains an array of tasks. Each task contains a course ID and page number.

**Comment by thomasplevy** [comment]
@pondermatic got it, thanks!

Are these piling up because the batches are erroring as a result of the #1598 ?

These duplicates don't appear to be the result of strings vs ints, although in reviewing `wp_get_next_scheduled()` I can see that the arguments are compared strictly so we do have a potential issue there (easily fixed with some casting) but I'd like to now

**Comment by thomasplevy** [comment]
@pondermatic it occurred to me that you may have been responding to my question about having any more issues related to the processor and not talking about the string/int issue here. If you have another issue let's move it into a new issue to discuss there.

Whether or not the above is related there is an issue with type handling and the way we're checking for whether or not the event is already scheduled. The #1601 addresses this.

**Comment by pondermatic** [comment]
> Are these piling up because the batches are erroring as a result of the #1598 ?

I believe the #1598 issue helped discover issue #1600 and #1602 .
